Functions/Admin/Backup-RsEncryptionKey.ps1

```diff
@@ -4,71 +4,134 @@
+	        Description
+	        -----------
+	        This command will back up the encryption key against named instance (SQL2012) from SQL Server 2012 Reporting Services
+	
+		.NOTES
+			Author:      ???
+			Editors:     Friedrich Weinmann
+			Created on:  ???
+			Last Change: 31.01.2017
+			Version:     1.1
+	
+			Release 1.1 (26.01.2017, Friedrich Weinmann)
+			- Fixed Parameter help (Don't poison the name with "(optional)", breaks Get-Help)
+			- Implemented ShouldProcess (-WhatIf, -Confirm)
+			- Replaced calling exit with throwing a terminating error (exit is a bit of an overkill when failing a simple execution)
+			- Improved error message on failure.
+			- Standardized the parameters governing the Report Server connection for consistent user experience.
+	
+			Release 1.0 (???, ???)
+			- Initial Release
     #>
 
-    [cmdletbinding()]
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
     param
     (
+        [Alias('SqlServerInstance')]
         [string]
-        $SqlServerInstance='MSSQLSERVER',
-
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
         [string]
-        $SqlServerVersion='13',
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential,
 
-        [Parameter(Mandatory=$True)]
+        [Parameter(Mandatory = $True)]
         [string]
         $Password,
         
-        [Parameter(Mandatory=$True)]
+        [Parameter(Mandatory = $True)]
         [string]
-        $KeyPath    
-    )
-
-    $rsWmiObject = New-RsConfigurationSettingObject -SqlServerInstance $SqlServerInstance -SqlServerVersion $SqlServerVersion
-
-    Write-Verbose "Retrieving encryption key..."
-    $encryptionKeyResult = $rsWmiObject.BackupEncryptionKey($Password)
-
-    if ($encryptionKeyResult.HRESULT -eq 0) 
-    {
-        Write-Verbose "Success!"
-    }
-    else 
-    {
-        Write-Error "Fail! `n Errors: $($encryptionKeyResult.ExtendedErrors)"
-        Exit 1
-    }
-
-    Write-Verbose "Writing key to file..."
-    [System.IO.File]::WriteAllBytes($KeyPath, $encryptionKeyResult.KeyFile)
-    Write-Verbose "Success!"
+        $KeyPath
+	)
+    
+    if ($ComputerName) { $tempComputerName = $ComputerName }
+    else { $tempComputerName = ([ReportingServicesTools.ConnectionHost]::ComputerName) }
+    if ($ReportServerInstance) { $tempInstanceName = $ReportServerInstance }
+    else { $tempInstanceName = ([ReportingServicesTools.ConnectionHost]::Instance) }
+        
+    if ($PSCmdlet.ShouldProcess("$tempComputerName \ $tempInstanceName", "Retrieve encryption key and create backup in $KeyPath"))
+	{
+		#region Connect to Report Server using WMI
+        try
+        {
+            $splat = @{ }
```

Comment:
If you're sure. Mind you: It's going to vanish with the refactoring, but I'll keep an eye open for any other $splats I find.
I usually go by splat since frankly: It's right above the function and the function's own name handles the purpose well enough IMO. Either way, it's a preference thing so I don't mind changing any $splats that remain.